### PR TITLE
CupertinoAlertDialog dark mode & CupertinoActionSheet fidelity

### LINKED ISSUE #20469: Correct CupertinoAlertDialog divider colors
After #18381 is merged in and is dedup'd against the similar Action Sheet work, figure out the right colors for the dividers when using the iOS barrier color (initially the dialog work was done against the material barrier color).

### LINKED ISSUE #39941: CupertinoActionSheet (and potentially CupertinoAlertDialog) fidelity issue
The color/opacity of the action items is off.

| Native | Current CupertinoActionSheet | Previous |
| --- | --- | --- |
| Simulator Screen Shot - iPhone 8 Plus - 2019-09-05 at 15 49 17 | image | image (3) |

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Description

- Adjust colors used in `CupertinoAlertDialog` and `CupertinoActionSheet`
- Change the `ModalBarrier` color that used for cupertino popup routes
- `CupertinoAlertDialog` dark mode.

## Related Issues

Part of https://github.com/flutter/flutter/issues/35541
Fixes https://github.com/flutter/flutter/issues/39941
Fixes #20469

## Screenshots (iPhone XR device, iOS 13.1 17A5831c)

The native alert view has some additional visual effects that we have yet to add to `CupertinoAlertDialog`, so it looks more opaque than `CupertinoAlertDialog` in the screenshots below.

| native | cupertino |
| --- | --- |
| image1 (1) | image0 |
|  image4 | image5 |
| image2 | image3 |
| image1 (2) | image0 (1) |

## Tests

I added the following tests:

- dialog dark mode
- route dark mode

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/route.dart [hunk 1]
```diff
@@ -947,7 +955,7 @@ Future<T> showCupertinoDialog<T>({
   return showGeneralDialog(
     context: context,
     barrierDismissible: false,
-    barrierColor: _kModalBarrierColor,
+    barrierColor: CupertinoDynamicColor.resolve(_kModalBarrierColor, context),
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/route.dart]: The caveat of doing this is, the color won't change unless the dialog is dismissed then shown again.

Comment (HansMuller) [reply packages/flutter/lib/src/cupertino/route.dart]: Seems OK.  Even if we recreated showGeneralDialog's _DialogRoute, it still wouldn't be easy to get the scrim to redraw when the platform's brightness changed.

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by flar** [comment]
This PR caused a regression as discussed in https://github.com/flutter/flutter/issues/50183

**Pull request merged**